kodilla-good-patterns/src/main/java/flightSearcher/FlightProcessor.java

```diff
@@ -1,26 +1,49 @@
 package flightSearcher;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class FlightProcessor {
 
-    CreateFlightList createFlightList;
-    FindFlight findFlight;
+    CreateFlightList createFlightList = new CreateFlightList();
 
-    public FlightProcessor(CreateFlightList createFlightList, FindFlight findFlight) {
-        this.createFlightList = createFlightList;
-        this.findFlight = findFlight;
+    public FlightProcessor() {
     }
 
-    public List<Flight> findFlightFrom(Flight flight) {
-        return findFlight.findFlightFrom(flight.getFrom());
+    List<Flight> inIt() {
+        List<Flight> temporaryList = createFlightList.getAllFlights();
+        return temporaryList;
     }
 
-    public List<Flight> findFlightTo(Flight flight) {
-        return findFlight.findFlightTo(flight.getTo());
+    public List<Flight> findFlightFrom(String flightFrom) {
+        List<Flight> flights = inIt();
+
+        return flights.stream()
+                .filter(flight1 -> flight1.getFrom().equals(flightFrom))
+                .collect(Collectors.toList());
     }
 
-    public List<FlightPair> findIndirect() {
-        return findFlight.findIndirectFlights();
+    public List<Flight> findFlightTo(String flightTo) {
+        List<Flight> flights = inIt();
+
+        return flights.stream()
+                .filter(flight1 -> flight1.getTo().equals(flightTo))
+                .collect(Collectors.toList());
+    }
+
+    public List<FlightPair> findIndirect(Flight flightFromTo) {
```

Comment:
Dla danych
Warszawa -> Kraków
Warszawa -> Londyn
Warszawa -> Gdańsk
Loty do Londynu:
Kraków -> Londyn
Warszawa -> Londyn
Gdańsk -> Londyn

Otrzymuje tym sposobem 
Warszawa -> Kraków + Warszawa -> Londyn
Warszawa -> Gdańsk + Warszawa -> Londyn

Czyli cos nie tak :)

Reply:
No nie, skoro jest informacja, że międzylądowanie w x albo y ale sam lot pozostaje: Warszawa > Londyn

Reply:
Ok, rozumiem gdzie miałem błąd, poprawiłem to String, teraz drukuje From-between + between -to